keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,920 @@
+
+In the alpha version of the Resource Manager applies driver plugins to manage resource
+allocations for Pods. Post-alpha we will consider if we need to explicitly mark which pods are to be handled by an allocation 
+plugin.  The overall architecture consists of a manager component – the Resource
+Manager inside the container manager, an optional Pod spec argument to identify which 
+Pods to be processed by the manager(post-alpha), a compute resource store – a component used to store
+the resource sets(cpusets, memory affinity configuration) for the resource plugins. The CCI Manager 
+is enabled via a feature gate. After turning on the feature gate all pod allocation requets for cpu 
+and memory will be handled by the CCI Manager.
+
+![image](CCIArchAlpha.jpg)<br>
+Figure 1: CCI Resource Manager Architecture inside Kubelet
+
+    Pod Spec “CCIDriverName Extension”
+    type PodSpec {
+       …
+    	 // CCIResourceDriver is an optional parameter which can identify if a Pod resource manager (cpu an memory)
+	    // has to be handled by a pluggable CCI resource driver 
+	    // This is an alpha field and requires enabling the
+    	 // CCIResourceManager feature gate.
+	    //
+    	 // This field is immutable.
+	    //
+	    // +featureGate=CCIResourceManager
+	    // +optional
+	    CCIResourceDriver string `js”n:”cciResourceDriver,omitem”ty” protob”f:”bytes,40,opt,name=cciResourceDri”er”`
+     …}
+
+The proposed extension relies on a new optional argument inside the Pod spec to `driverName`. 
+We use this optional argument to drive the allocation process. If the driver name is not provided,
+the Pod resources will be allocated through the standard CPU manager within t ubeletlet. An
+alternative association approach can be considered during the implementation which can avoid
+Pod spec changes.
+
+1. CCI Resource Manager<br>
+These changes will need a separate resource manager component inside the container 
+manager which will handle plugin management tasks, including plugin registration and 
+invocations of registered plugins.  These tasks will be managed via grpc for Pods who 
+require one or more resource management plugin(s) to get successfully allocated.  In 
+the alpha version we provide to entry points to plugins on admission,  addContainer 
+and removeContainer lifetime events (plugins have to implement Admit, AddResource 
+and RemoveResource handlers accordingly). 
+
+2. CCI Resource Store<br>
+The Resource Manager uses a store to keep track of resourcesets which can include 
+cpusets, memory, etc ..  allocated by CCI Drivers. This information is gathered 
+per container. The store will be used by a new cpu manager policy to synchronize 
+the state between cpu manager and pluggable CCI Drivers.  The store offers the 
+following interface to manage resource sets.
+
+        +AddResource(Pod, container, resourceset)
+        +RemoveResource(container)
+        +AvailablResources(): resourceset
+
```

Comment:
i added clarification that we use a map with pod id and container name to store the resource sets. Initially a resource set will consist of a cpuset. We will add memory in later stage. the map will be serialized together with the list of available resources. This could enable the restart functionality. I added that as doc.